### ISSUE TITLE
re-export immer utilities

### ISSUE THREAD

**Issue by shadeofgod** [reduxjs/redux-toolkit#850]
Hey guys, I see there is only `current` function re-exported from immer. Is it possiple to re-export other utilities like `immerable `, `original` etc...?

also I want to have patches after state change for debugging purpose, immer provides `produceWithPatches` or `enablePatches` and `applyPatches` if you pass a third argument to `produce` function to collect the patches. How can I acheive that with redux-toolkit?

**Comment by phryneas** [comment]
Afaik, we are currently exporting `current` and `original`.

But actually, due to the fact that immer tree-shakes, every new re-export from RTK will actually increase the size of RTK - the patch stuff will add about 1kb, so I'd prefer you just directly import from immer.

**Comment by shadeofgod** [comment]
Yeah import from immer directly is doable.

But for patches there seems no way to replace the produce call inside RTK?

**Comment by phryneas** [comment]
How would you access these patches? The reducer is called by redux, the return value from that call is the new state.

**Comment by markerikson** [comment]
Yeah, not something we plan to do at this point (or really can do given RTK's design).

**Comment by eduardomoroni** [comment]
Hey, sorry for reviving such an old thread. I wonder if there's a possibility to add this behaviour as a redux middleware or middleware listener.

I noticed that RTK query exposed similar api
https://redux-toolkit.js.org/rtk-query/api/created-api/api-slice-utils#description

**Comment by EskiMojo14** [comment]
not sure what behaviour you're asking for, could you be more specific?

**Comment by eduardomoroni** [comment]
Being able to access somehow immer patches, with RTK query you have a way to access them.

**Comment by phryneas** [comment]
But RTK query is an optional add-on, we do not ship that with the core.

**Comment by eduardomoroni** [comment]
> But RTK query is an optional add-on, we do not ship that with the core.
 Yeah sorry for the confusion, I didn't mean to add to the library. I was asking if there would be a way of accessing it.
 
 @EskiMojo14 Thanks for the suggestion. Yeah I was looking for something like this, I will check how I could make this "global" in my case I kind of need for all the actions.
